contracts/SponsorFaucet.sol

```diff
@@ -331,26 +276,22 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard, WhitelistedRole {
     }
 
     function _validateApplyForCollateral(address dapp) internal {
-        require(
-            address(this).balance >= collateral_bound,
-            "ERROR_COLLATERAL_FAUCET_OUT_OF_MONEY"
-        );
         uint256 collateral_balance = internal_sponsor
             .getSponsoredBalanceForCollateral(dapp);
-        uint256 contract_collateral_bound;
-        uint256 contract_collateral_total_limit;
-        
-        if (isGeneralContract(dapp)) {
-            contract_collateral_bound = general_collateral_bound;
-            contract_collateral_total_limit = general_collateral_total_limit;
-        } else if (isSpecialContract(dapp)) {
-            contract_collateral_bound = special_dapps[dapp].collateral_bound;
-            contract_collateral_total_limit = special_dapps[dapp].collateral_total_limit;
-        } else {
-            contract_collateral_bound = collateral_bound;
-            contract_collateral_total_limit = collateral_total_limit;
+        uint256 contract_collateral_bound = dapp_bounds[small].collateral_bound;
+        uint256 contract_collateral_total_limit = dapp_bounds[small].collateral_total_limit;
```

Comment:
These are duplicated codes, how about to add a `private` method to simplify code? E.g. getBounds(addr)

Reply:
internal view method added